GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+# Governance documentation
+
+*Disclaimer: This is work in progress. Mechanisms are susceptible to change.*
+
+This document describes the high-level architecture of the governance module. The governance module allows bonded Atom holders to vote on proposals on a 1 bonded Atom 1 vote basis.
+
+## Design overview
+
+The governance process is divided in a few steps that are outlined below:
+
+- **Proposal submission:** Proposal is submitted to the blockchain with a deposit 
+- **Vote:** Once deposit reaches a certain value (`MinDeposit`), proposal is confirmed and vote opens. Bonded Atom holders can then send `TxVote` transactions to vote on the proposal
+- If the proposal involves a software upgrade
+  - **Signal:** Validator start signaling that they are ready to switch to the new version
+  - **Switch:** Once more than 2/3rd validators have signaled their readiness to switch, their software automatically flips to the new version
+  
+## Proposal submission
+
+### Right to submit a proposal
+
+Any Atom holder, whether bonded or unbonded, can submit proposals by sending a `TxProposal` transaction. Once a proposal is submitted, it is identified by its unique `proposalID`.
+
+### Proposal filter (minimum deposit)
+
+To prevent spam, proposals must be submitted with a deposit in Atoms. Voting period will not start as long as the proposal's deposit is smaller than the minimum deposit parameter `MinDeposit`.
+
+When a proposal is submitted, it has to be accompagnied by a deposit that must be strictly positive but that can be inferior to `MinDeposit`. Indeed, the submitter need not pay for the entire deposit on its own. If a proposal's deposit is strictly inferior to `MinDeposit`, other Atom holders can increase the proposal's deposit by sending a `TxDeposit` transaction. Once the proposals's deposit reaches `minDeposit`, it enters voting period. 
+
+### Deposit refund
+
+There are two instances where Atom holders that deposited can claim back their deposit:
+- If the proposal is accepted
+- If the proposal's deposit does not reach `MinDeposit` for a period longer than `mMxDepositPeriod` (initial value: 2 months). Then the proposal is considered closed and nobody can deposit on it anymore.
+
+In such instances, Atom holders that deposited can send a `TxClaimDeposit` transaction to retrieve their share of the deposit.
+
+### Proposal types
+
+In the initial version of the governance module, there are two types of proposal:
+- `PlainTextProposal`. All the proposals that do not involve a modification of the source code go under this type. For example, an opinion poll would use a proposal of type `PlainTextProposal`
```

Comment:
What is an opinion poll in this context? I can see proposals which outline the intention to upgrade the protocol in a certain way whichout actually providing the code... is this what you mean? we should go into greater detail on this idea

Reply:
> I can see proposals which outline the intention to upgrade the protocol in a certain way whichout actually providing the code...

Yes good example. Actually `PlainTextProposals` encompass everything that is not an immediate proposition to update the protocol. Another example could be "Scaling Tendermint: roadmap A or roadmap B?" (like Casper FFG vs Casper CBC)